src/Interfaces/interface.ts

```diff
@@ -0,0 +1,38 @@
+import { Dispatch } from 'react';
+import { TodoAction } from './actions';
+
+export type TodoItem = {
+  id: number;
+  content: string;
+  done: boolean;
+};
+
+export type TodoList = TodoItem[];
+
+export interface ITodoCtx {
+  todoList: TodoList;
+  todoDispatch: Dispatch<TodoAction>;
+}
+
+export interface ITodoItemProps {
+  item: TodoItem;
+}
+
+export interface ITodoContainerProps {
+  type: 'To-Do' | 'Done';
+  items: TodoList;
+}
+
+export type IInitValueArg<T> = T;
+
+export type IUseInputArg = IInitValueArg<string>;
+
+export interface ITextNeedFnArg {
+  (text: string): void;
+}
+
+export interface IIdNeedFnArg {
+  (id: number): void;
+}
+
+export type TodoContextHandler = ITextNeedFnArg;
```

Comment:
```suggestion
export type TodoContextHandler = (text:string) => void;
```

이 부분은 불필요한 타이핑인 것 같습니다.

또, 네이밍에서는 Arg를 붙여주셨으나 실제 타입은 함수 타입이라서 혼동이 있을 수 있을 것같네요

Reply:
급하게 과제 와라락 해버린 티가 나네요...ㅎ